### PR TITLE
[IN-8] Guard clauses implementation

### PR DESCRIPTION
**Pull request by glachezar**

### CODE REVIEW

File: investment-app/TRINV.Api/TRINV.Domain/Validations/Ensure.cs [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    public static void IsNull(object? value,
+        [CallerArgumentExpression("value")] string? paramName = null)
+    {
+        if (value is null)
+        {
+            throw new ArgumentException("Object value can't be null or empty", paramName);
+        }
+    }
+
+    public static void IsEqual(
```
Comment (MihaelPavlov) [inline investment-app/TRINV.Api/TRINV.Domain/Validations/Ensure.cs:44]: As we discussed. Did you checkl how IsEqual is working with object and primitive types

Comment (glachezar) [reply investment-app/TRINV.Api/TRINV.Domain/Validations/Ensure.cs]: Done.

File: investment-app/TRINV.Api/TRINV.Domain/Validations/Ensure.cs [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    {
+        if (value is null)
+        {
+            throw new ArgumentException("Object value can't be null or empty", paramName);
+        }
+    }
+
+    public static void IsEqual<T>(T? value1, T? value2)
+    {
+        var value1Type = value1.GetType();
```
Comment (MihaelPavlov) [inline investment-app/TRINV.Api/TRINV.Domain/Validations/Ensure.cs:37]: We can simplify this code here in IsEqual. 
Get the type just one time .
The type of the both objects will be the same. The same for the properties.
And you don't need to compare the propertyName . We need just ther value.

Here check this one: More simplified way: 

` 

        Type type = typeof(T);
        foreach (var field in type.GetFields())
        {
            object value1 = field.GetValue(obj1);
            object value2 = field.GetValue(obj2);

            if (value1 == null && value2 == null) continue;
            if (value1 == null || !value1.Equals(value2)) return false;
        }
        return true;

`

File: investment-app/TRINV.Api/TRINV.Domain/Validations/Ensure.cs [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+using System.Text.RegularExpressions;
+
+public static class Ensure
+{
+    public static void IsArgumentNullOrEmpty(string? value,
+        [CallerArgumentExpression("value")] string? paramName = null)
+    {
+        if (string.IsNullOrEmpty(value))
+        {
+            throw new ArgumentException("The argument value can't be null or empty", paramName);
```
Comment (MihaelPavlov) [inline investment-app/TRINV.Api/TRINV.Domain/Validations/Ensure.cs:13]: Sorry now i notice that the current methods should be boolean. We don't want to interrupt the flow of the request with exceptions. Please make them all bollean. return true and false;

Comment (glachezar) [reply investment-app/TRINV.Api/TRINV.Domain/Validations/Ensure.cs]: Done.

**Review by MihaelPavlov — APPROVED** [review]

**Pull request merged**